### ISSUE TITLE
Build failure: generator raised StopIteration

### ISSUE THREAD

**Issue by mxmilkiib** [PatrickStephansen/live-cue#1]
```
 ~/src/live-cue master                                                                                                  17:32:41
> ./waf configure
Traceback (most recent call last):
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Node.py", line 295, in ant_iter
    raise StopIteration
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Scripting.py", line 122, in waf_entry_point
    run_commands()
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Scripting.py", line 179, in run_commands
    parse_options()
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Scripting.py", line 152, in parse_options
    Context.create_context('options').execute()
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Options.py", line 149, in execute
    super(OptionsContext,self).execute()
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Context.py", line 88, in execute
    self.recurse([os.path.dirname(g_module.root_path)])
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Context.py", line 129, in recurse
    user_function(self)
  File "/home/milk/src/live-cue/wscript", line 13, in options
    opt.load('compiler_c')
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Context.py", line 85, in load
    fun(self)
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Tools/compiler_c.py", line 40, in options
    opt.load_special_tools('c_*.py',ban=['c_dumbpreproc.py'])
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Context.py", line 319, in load_special_tools
    lst=self.root.find_node(waf_dir).find_node('waflib/extras').ant_glob(var)
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Node.py", line 344, in ant_glob
    ret=[x for x in self.ant_iter(accept=accept,pats=[to_pat(incl),to_pat(excl)],maxdepth=kw.get('maxdepth',25),dir=dir,src=src,remove=kw.get('remove',True))]
  File "/home/milk/src/live-cue/.waf3-1.9.12-46ae1733752575b7ea2378b1eca4e1ae/waflib/Node.py", line 344, in <listcomp>
    ret=[x for x in self.ant_iter(accept=accept,pats=[to_pat(incl),to_pat(excl)],maxdepth=kw.get('maxdepth',25),dir=dir,src=src,remove=kw.get('remove',True))]
RuntimeError: generator raised StopIteration
```
I'm on Arch Linux.

**Comment by PatrickStephansen** [comment]
It looks like it fails right on the first line of the script. I think this waf version is old enough that it expects python 2. I've updated the waf version so it should at least compile for you now with python3.

You're the first other person to try this. I have to warn you that this was made for an extremely niche use-case and I stopped working on it as soon as it was just good enough for me to use. The list of tracks is hard-coded in the c source file, so you will also need to replace that with something that actually exists on your computer before it will compile to something that will be able to run. I've only ever run it with jalv (using command `jalv http://github.com/PatrickStephansen/live-cue`), so I don't know if it even works with other jack hosts. It should work with mono or stereo wav files.

It takes a mono audio input and starts playing a stereo sample from the hard-coded playlist when the input clips (I never got the input threshold to work so it has to clip). If the input clips again while the sample is playing, it stops and queues the next track. If the sample plays to the end it also queues the next track. I use this to trigger samples with a drum pad connected straight to my laptop with no processing in between, usually straight to usb with a Rocksmith cable.
